h/static/scripts/group-forms/hooks/use-update-moderation-status.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useContext, useEffect, useRef } from 'preact/hooks';
+
+import { Config } from '../config';
+import type { APIAnnotationData, ModerationStatus } from '../utils/api';
+import { callAPI } from '../utils/api';
+
+/**
+ * Returns a callback that can be used to update the moderation status of an
+ * annotation via API call
+ */
+export function useUpdateModerationStatus(annotation: APIAnnotationData) {
+  const config = useContext(Config);
+
+  // Track the last in-flight request
+  const abortCtrlRef = useRef<AbortController>();
+  useEffect(() => {
+    const abortCtrl = abortCtrlRef.current;
+    return () => abortCtrl?.abort();
+  }, []);
```

Comment:
Hmm... aborting the request could potentially cause a surprising behavior if you cause an annotation card to be unmounted (eg. by changing the filter) while being saved. Depending where the request is in the pipeline it may or may not actually get canceled. We might be better off just leaving the request to complete if in-flight.